### PR TITLE
create evaluation task according to worker num

### PR DESCRIPTION
**Pull request by QiJune**

### CODE REVIEW

File: elasticdl/python/master/task_dispatcher.py [hunk 1]
```diff
@@ -64,57 +65,104 @@ class _TaskDispatcher(object):
...
-            evaluation_shards: A dictionary from RecordIO file name to
-                the number of evaluation records.
-            prediction_shards: A dictionary from RecordIO file name to
-                the number of prediction records.
-            records_per_task: The number of records per task.
-            num_epochs: The total number of epochs for the tasks where
-                an epoch is a complete iteration over the shards.
-            callbacks_list: The Keras CallbacksList object to contain all
-                callback instances.
-        """
```
Comment (workingloong) [inline elasticdl/python/master/task_dispatcher.py:87]: Why do you remove the docstring?

File: elasticdl/python/master/task_dispatcher.py [hunk 2]
```diff
@@ -64,57 +65,104 @@ class _TaskDispatcher(object):
...
+                ).create_shards(),
+                self._training_records_per_task,
+            )
+        else:
+            if self._eval_table_size == 0:
+                reader = self._create_data_reader_fn(
+                    data_origin=self._validation_data, partition=partition,
+                )
+                if hasattr(reader, "get_table_size") and callable(
+                    reader.get_table_size
```
Comment (terrytangyuan) [inline elasticdl/python/master/task_dispatcher.py:132]: This is only applicable to readers that has the concept of "table"

**Review by terrytangyuan — CHANGES_REQUESTED** [review]

File: elasticdl/python/data/reader/odps_reader.py [hunk 3]
```diff
@@ -61,6 +61,11 @@ def records_output_types(self):
     def metadata(self):
         return self._metadata
 
+    def get_table_size(self):
+        check_required_kwargs(["table"], self._kwargs)
+        reader = self._get_reader(self._kwargs["table"])
```
Comment (chunyang-wen) [inline elasticdl/python/data/reader/odps_reader.py:66]: It is better we cache the result and reuse it.

File: elasticdl/python/master/task_dispatcher.py [hunk 4]
```diff
@@ -64,57 +65,104 @@ class _TaskDispatcher(object):
...
-        elif self._evaluation_shards:
+        elif self._validation_data:
             self.create_tasks(elasticdl_pb2.EVALUATION)
-        elif self._prediction_shards:
+        elif self._prediction_data:
             self.create_tasks(elasticdl_pb2.PREDICTION)
 
+    def _create_shards(self, task_type):
+        kwargs = get_dict_from_params_str(self._data_reader_params)
+        partition = kwargs.get("partition", None) if kwargs else None
```
Comment (chunyang-wen) [inline elasticdl/python/master/task_dispatcher.py:113]: `kwargs` is a dict? If so,

```python
partition = kwargs.get("partition", None)
```

**Pull request closed without merging**